src/main/java/com/ibm/kappnav/logging/Logger.java

```diff
@@ -30,73 +30,87 @@
+    
+    public static boolean isDebugEnabled() { 
+        return typeEnabled[LogType.DEBUG.ordinal()]; 
+    }
+
    private static void setLogTypes(boolean value) { 
       for ( LogType type: LogType.values() ) { 
-          enabled[type.ordinal()]= value; 
+          typeEnabled[type.ordinal()]= value; 
       } 
    } 
 
-   public static void setLogLevel(String level) {
-            level= level.toLowerCase(); 
+   public static void setLogLevel(LogLevel level) {
             setLogTypes(false); 
             switch(level) { 
-                case "none": 
+                case NONE: 
                     break;               
-                case "error":
-                    enabled[LogType.ERROR.ordinal()]= true;  
+                case ERROR:
+                    typeEnabled[LogType.ERROR.ordinal()]= true;  
                     break;
-                case "warning": 
-                    enabled[LogType.ERROR.ordinal()]= true;  
-                    enabled[LogType.WARNING.ordinal()]= true;  
+                case WARNING: 
+                    typeEnabled[LogType.ERROR.ordinal()]= true;  
+                    typeEnabled[LogType.WARNING.ordinal()]= true;  
                     break;
-                case "info": 
-                    enabled[LogType.ERROR.ordinal()]= true;  
-                    enabled[LogType.WARNING.ordinal()]= true;  
-                    enabled[LogType.INFO.ordinal()]= true;  
+                case INFO: 
+                    typeEnabled[LogType.ERROR.ordinal()]= true;  
+                    typeEnabled[LogType.WARNING.ordinal()]= true;  
+                    typeEnabled[LogType.INFO.ordinal()]= true;  
                     break;
-                case "entry": 
-                    enabled[LogType.ERROR.ordinal()]= true;  
-                    enabled[LogType.WARNING.ordinal()]= true;  
-                    enabled[LogType.INFO.ordinal()]= true;  
-                    enabled[LogType.ENTRY.ordinal()]= true;  
-                    enabled[LogType.EXIT.ordinal()]= true;  
+                case DEBUG: 
+                    typeEnabled[LogType.ERROR.ordinal()]= true;  
+                    typeEnabled[LogType.WARNING.ordinal()]= true;  
+                    typeEnabled[LogType.INFO.ordinal()]= true;  
+                    typeEnabled[LogType.ENTRY.ordinal()]= true;  
+                    typeEnabled[LogType.EXIT.ordinal()]= true;  
                     break;
-                case "debug": 
+                case ENTRY: 
                     setLogTypes(true); 
                     break;
-                case "all": 
+                case ALL: 
                     setLogTypes(true); 
                     break;
                 default : // same as info
-                    enabled[LogType.ERROR.ordinal()]= true;  
-                    enabled[LogType.WARNING.ordinal()]= true;  
-                    enabled[LogType.INFO.ordinal()]= true;  
+                    typeEnabled[LogType.ERROR.ordinal()]= true;  
```

Comment:
With the switch to using LogLevel instead of String there's no way to have an invalid value, so we no longer need the default case.  No need to update for this PR.  Please just keep it in mind for the next one.

Reply:
OK thank Roger